Samples/WinMLSamplesGallery/TransformAsync/SegmentModel.cpp

```diff
@@ -0,0 +1,279 @@
+
+	m_outputVideoFrame.CopyToAsync(outVideoFrame).get();
+
+	m_syncStarted = FALSE;
+}
+
+LearningModel StyleTransfer::GetModel()
+{
+	auto model_path = std::filesystem::current_path();
+	model_path.append("Assets\\mosaic.onnx");
+	return LearningModel::LoadFromFilePath(model_path.c_str());
+}
+
+/****	Background blur model	****/
+BackgroundBlur::~BackgroundBlur() 
+{
+	if (m_session) m_session.Close();
+}
+
+void BackgroundBlur::InitializeSession(int w, int h)
+{
+	w /= m_scale; h /= m_scale;
+	SetImageSize(w, h);
+
+	auto joinOptions1 = LearningModelJoinOptions();
+	joinOptions1.CloseModelOnJoin(true);
+	joinOptions1.Link(L"Output", L"input");
+	joinOptions1.JoinedNodePrefix(L"FCN_");
+	joinOptions1.PromoteUnlinkedOutputsToFusedOutputs(true);
+	auto modelExperimental1 = LearningModelExperimental(Normalize0_1ThenZScore(h, w, 3, m_mean, m_stddev));
+	LearningModel intermediateModel = modelExperimental1.JoinModel(GetModel(), joinOptions1);
+
+	auto joinOptions2 = LearningModelJoinOptions();
+	joinOptions2.CloseModelOnJoin(true);
+	joinOptions2.Link(L"FCN_out", L"InputScores");
+	joinOptions2.Link(L"OutputImageForward", L"InputImage");
+	joinOptions2.JoinedNodePrefix(L"Post_");
+	//joinOptions2.PromoteUnlinkedOutputsToFusedOutputs(false); // TODO: Causes winrt originate error in FusedGraphKernel.cpp, but works on CPU
+	auto modelExperimental2 = LearningModelExperimental(intermediateModel);
+	LearningModel modelFused = modelExperimental2.JoinModel(PostProcess(1, 3, h, w, 1), joinOptions2);
+
+	// Save the model for debugging purposes
+	//modelExperimental2.Save(L"modelFused.onnx");
+
+	m_session = CreateLearningModelSession(modelFused);
+	m_binding = LearningModelBinding(m_session);
+}
+LearningModel BackgroundBlur::GetModel()
+{
+	auto model_path = std::filesystem::current_path();
+	model_path.append("Assets\\fcn-resnet50-12.onnx");
+	return LearningModel::LoadFromFilePath(model_path.c_str());
+}
+
+void BackgroundBlur::Run(IDirect3DSurface src, IDirect3DSurface dest)
+{
+	m_syncStarted = TRUE;
+
+	VideoFrame inVideoFrame = VideoFrame::CreateWithDirect3D11Surface(src);
+	VideoFrame outVideoFrame = VideoFrame::CreateWithDirect3D11Surface(dest);
+	SetVideoFrames(inVideoFrame, outVideoFrame);
+
+	// Shape validation
+	assert((UINT32)m_inputVideoFrame.Direct3DSurface().Description().Height == m_imageHeightInPixels);
+	assert((UINT32)m_inputVideoFrame.Direct3DSurface().Description().Width == m_imageWidthInPixels);
+
+	hstring inputName = m_session.Model().InputFeatures().GetAt(0).Name();
+	hstring outputName = m_session.Model().OutputFeatures().GetAt(1).Name();
+
+	m_binding.Bind(inputName, m_inputVideoFrame);
+	m_binding.Bind(outputName, m_outputVideoFrame);
+	auto results = m_session.Evaluate(m_binding, L"");
+	m_outputVideoFrame.CopyToAsync(outVideoFrame).get();
+	m_syncStarted = FALSE;
+}
+
+LearningModel BackgroundBlur::PostProcess(long n, long c, long h, long w, long axis)
+{
+	auto builder = LearningModelBuilder::Create(opset)
```

Comment:
>LearningModelBuilder [](http://example.com/codeflow?start=16&length=20)

Wow, I didn't even know this existed. Oi, it appears it would be so easy to make a mistake 😬,

Reply:
it is indeed very easy to make a mistake! it's mostly a matter of pattern recognition with the onnx operator docs